src/services/stringCompletions.ts

```diff
@@ -659,11 +683,15 @@ namespace ts.Completions.StringCompletions {
 
         const pathPrefix = path.slice(0, path.length - 1);
         const remainingFragment = tryRemovePrefix(fragment, pathPrefix);
-        return remainingFragment === undefined ? justPathMappingName(pathPrefix) : flatMap(patterns, pattern =>
-            getModulesForPathsPattern(remainingFragment, baseUrl, pattern, fileExtensions, host));
+        if (remainingFragment === undefined) {
+            const starIsFullPathComponent = path[path.length - 2] === "/";
+            return starIsFullPathComponent ? justPathMappingName(pathPrefix) : flatMap(patterns, pattern =>
+                getModulesForPathsPattern("", baseUrl, pattern, fileExtensions, host)?.map(({ name, ...rest }) => ({ name: pathPrefix + name, ...rest })));

```

Comment:
Pedantic: can you get away with `pathPrefix + name` or do you need to use `combinePaths`?

Reply:
It’s explicitly not combining paths; these might be two halves of a single directory name.